### ISSUE TITLE
webapp: "show all runs for a given commit" is broken

### ISSUE THREAD

**Issue by austin3dickey** [conbench/conbench#918]
Previously, `benchalerts` was able to link to a table in the webapp which showed all runs associated with a given commit. It accomplished this by linking to `<conbench_url>/?search=<commit_hash>`. That was a little hacky, but it worked well. Now that the table-wide search bar does not anchor the URL, this view does not work anymore.

**Comment by jgehrcke** [comment]
> <conbench_url>/?search=<commit_hash>

Gotcha! This wasn't covered by tests (:fire:) , and there was no insightful code commentary around this. It's impressive that we used this for building a "show all runs for a given commit" view.

Of course we have to address this. Let's maybe take a moment to see if we want to re-do `/?search=<commit_hash>` or use the opportunity to do it a tiny bit cleaner (if you're open to adjusting benchalerts then).

**Comment by austin3dickey** [comment]
I'm definitely open to adjusting benchalerts later! The use case is simply to power the "show more details" link at the bottom of a GitHub Check like [this one](https://github.com/conbench/benchalerts/runs/12158432127). This will allow users to click through for more details about all runs the report might be talking about.

In the short term, I could link to each individual run's page in the report, and just use the landing page for the "more details" link. But let's think about the ideal state.

**Comment by jonkeane** [comment]
Another use case: sharing a filtered view. One example: I want to show off all of the ursa-i9 hardwares from https://conbench.ursa.dev/ I used to be able to search for that (either in the omni search or the column filters) and it would update the URL so I could share that filtered view

**Comment by jgehrcke** [comment]
We might want to distinguish two different features:

1) Introduce a special view for "show all runs for a given commit", with a definite URL path pattern
2) A generic "maintain table search state in the location bar" feature (which should work in all views that show one or more datatables)

> Another use case: sharing a filtered view.

I think that is (2). Yes, we absolutely want to have that. I was thinking the same when I added this comment here:

https://github.com/conbench/conbench/pull/891/files#diff-a204bf4fa0ff5008526556829c2234be21e7bf0da0a1c6959976213c539bf471R124

I created a specific ticket for that: https://github.com/conbench/conbench/issues/923

-----

For addressing this ticket here: do we want (1) or (2)? It looks like we want to build (2) anyway. So maybe we do that first.

But when I then think about how we would use (2) for this here: we would (mis) use the landing page for this kind of lookup, and that would be limited to searching through last 1000 runs -- is that good?

**Comment by jonkeane** [comment]
Thanks for separating these out, agreed that they are slightly different + most importantly have slightly different implementations | requirements.

> we would (mis) use the landing page for this kind of lookup, and that would be limited to searching through last 1000 runs -- is that good?

Agreed that ultimately this misuse is not good. It might buy us a little bit of time. Especially if we add more details to the benchalerts report like direct links to each run anyway (so one _can_ get to them if they want to) that's enough for now. But agreed if we want a proper from-this-commit to the runs that it's associated with, we will need a slightly different UI | endpoint

**Comment by austin3dickey** [comment]
Agreed with both of you - the hack to use the anchored landing page wasn't ideal because it was only limited to the last 1000 runs.

Soon, I'll add direct links to all runs in the `benchalerts` reports, and link to the (unanchored) landing page in the "view more details" link. Those direct links will be really nice to have in the reports anyway. So we won't need to worry about feature (1) right now. Frankly, I'm not sure feature (1) will be worth the effort to implement for a long time.

**Comment by jgehrcke** [comment]
> Those direct links will be really nice to have in the reports anyway. 

Right, the existing  `https://<conbench-dns>/runs/<run-id>` seems to be useful for a report.

What do our current users need "show all runs for a given commit" for, today? Was that 'just' a nice thing to link to, or is this a critical lookup? (in view of the alternative of linking to specific runs directly)

**Comment by austin3dickey** [comment]
That was just a "nice" thing to link to, since we have to choose one link for the "view more details" button at the bottom of a GitHub Check. And I didn't want to only choose one run or benchmark result for that link. I'll switch that to just `https://<conbench-dns>`.
